pkg/sql/plan/build_delete.go

```diff
@@ -55,7 +55,8 @@ func buildDelete(stmt *tree.Delete, ctx CompilerContext, isPrepareStmt bool) (*P
 	beginIdx := 0
 	// needLockTable := !tblInfo.isMulti && stmt.Where == nil && stmt.Limit == nil
 	// todo will do not lock table now.
-	needLockTable := false
+	isDeleteWithoutFilters := stmt.Where == nil && len(tblInfo.tableDefs) == 1
```

Comment:
should like:
isDeleteWithoutFilters := !tblInfo.isMulti && stmt.Where == nil && stmt.Limit == nil